doc/elasticdl_on_sqlflow.md

```diff
@@ -0,0 +1,141 @@
+# _Design:_ ElasticDL on sqlflow
+
+## Overview
+
+This is a design doc on integration with [ElasticDL](https://github.com/wangkuiyi/elasticdl).
+
+### User Interface
+
+#### Training Job Submission
+
+```sql
+SELECT 
+    c1, c2, c3, c4, c5 as class
+FROM training_data
+TRAIN ElasticDLKerasClassifier
+WITH
+  optimizer = "optimizer"
+  loss = "loss"
+  eval_metrics = "eval_metrics_fn"
+  num_classes = 10
+COLUMN
+  c1,
+  NUMERIC(c2, 10),
+  BUCKET(c3, [0, 10, 100]),
+  c4
+LABEL class
+INTO trained_elasticdl_keras_classifier;
+```
+
+#### Prediction Job Submission
+
+```sql
+SELECT 
+    c1, c2, c3, c4
+FROM prediction_data
+PREDICT prediction_results_table
+WITH
+  num_classes = 10
+USING trained_elasticdl_keras_classifier;
+```
+
+#### Run-time Configurations
+
+Users can provide run-time configurations to ElasticDL job via additional parameters with prefix "runtime" within `WITH` clause, for example:
+
+```sql
+SELECT
+    c1, c2, c3, c4, c5 as class
+FROM training_data
+TRAIN ElasticDLKerasClassifier
+WITH
+  optimizer = "optimizer"
+  loss = "loss"
+  eval_metrics = "eval_metrics_fn"
+  num_classes = 10
+  runtime.num_epochs = 2
+  runtime.master_resource_request = "cpu=400m,memory=1024Mi"
+  runtime.master_resource_limit = "cpu=400m,memory=1024Mi"
+  runtime.worker_resource_request = "cpu=400m,memory=2048Mi"
+  runtime.worker_resource_limit = "cpu=1,memory=3072Mi"
+  runtime.records_per_task = 100
+  runtime.num_workers = 2
+COLUMN
+  c1, c2, c3, c4
+LABEL class
+INTO trained_elasticdl_keras_classifier;
+```
+
+### Implementation Details
+
+#### Training Job
+
+Steps:
+
+1. Based on `SELECT ... FROM ...`, read ODPS table and write it to [RecordIO](https://github.com/wangkuiyi/recordio) files, including both features and labels.
```

Comment:
Where should we store the RecordIO file? Why not reading from the ODPS table directly?

Reply:
For now we have to convert them to RecordIO files. They will be stored in Kubernetes persistent volumes.

Reply:
@tonyyang-svail Support reading directly from ODPS will be added later, after we complete the entire SQLFlow->ElasticDL work.